May 2023 Sprint

wwelling opened the pull request.

tamu-sad-iii committed:
Update asideTemplate.html (#281)

tamu-sad-iii committed:
Improve date conversion while build data network (#283)

* Cast as date, or parse year from iso string

else not counted

* Cleanup debug logging

* Increase readability

* Improve data network date format using utility

TODO: date format utility needs unit tests

* Some cleanup

* Update IndividualRepo.java

tamu-sad-iii committed:
Fix starts with filter query (#288)

Remove some dead code, probably again.

tamu-sad-iii committed:
Fix goof-up in discovery facet page (#299)

* Fix goof up in discovery facet page

* Add comment for page number

tamu-sad-iii committed:
Add ORCID to person discovery view csv export (#292)

tamu-sad-iii committed:
Add grant discovery view contributor organization facet (#293)

* Harvest, index, and expose grant contributor organization facet

* Add missing prefix rdfs to contributor organization sparql

* Add missing prefix foaf to contributor organization prefix

* Update contributorOrganization.sparql

* Update discoveryViews.yml

tamu-sad-iii committed:
Update asideTemplate.html (#302)

tamu-sad-iii committed:
Zip export of docx from an individual or individual's multiple reference (#300)

* Minor gitignore update

* Minor template formatting

* Filter query builder does not support contain

* Zipped Docx Exporter

Abstraction or utility needed for cleaning up redundant code copied

* Have Solr filter and sort lazy references by ids

Using JsonQueryRequest to avoid limitation of URL length when batch of ids increase to large.

* Apply filter and sorting to Solr JSON query request

* Update export view sort orders

* Add profiles to applicaiton.yml

* Pass limit as rows to lazy references Solr query

* Abstract docx redundancies

* Default max limit for export references

* Add helper for current year

* Add additional organization export

* Update department publication templates

* Update ExportFieldViewTest.java

* Set individual type conditionally on Solr doc type is defined

* Zip individual docx if no multiple reference provided

* Correct some template paths

* Add hyphen before timestamp of zip entry

* Update IndividualRepo.java

* Update .gitignore

* Use constant

* Update .gitignore

* Make utility constructor private

* Move zip file to utility

* Update DateFormatUtility.java

tamu-sad-iii committed:
Replacing + with space in angular route (#303)

rmathew1011 commented on src/main/java/edu/tamu/scholars/middleware/discovery/model/repo/IndividualRepo.java:
```diff
@@ -52,12 +56,14 @@
 public class IndividualRepo implements IndexDocumentRepo<Individual> {
 
-  private static final Logger logger = LoggerFactory.getLogger(IndividualRepo.class);
+    private static final Logger logger = LoggerFactory.getLogger(IndividualRepo.class);
```
Exrta space here

wwelling replied:
<img width="390" alt="image" src="https://github.com/TAMULib/scholars-discovery/assets/8352733/6379a4a8-29ad-4be7-8527-9cd69f2a7a7a">

wwelling replied:
LGTM

tamu-sad-iii committed:
Cleanup - repurposing branch (#307)

rmathew1011 commented on src/main/java/edu/tamu/scholars/middleware/messaging/handler/CustomStompSubProtocolErrorHandler.java:
```diff
@@ -12,9 +12,9 @@ public class CustomStompSubProtocolErrorHandler extends StompSubProtocolErrorHan
-    	}
+        if (Objects.nonNull(clientHeaderAccessor)) {
+            errorHeaderAccessor.setDestination(clientHeaderAccessor.getDestination());
+        }
```
This is unusual request - the indentation  on Lines 15 and 17 are highlighted - although only Line 16 has the intended change.

wwelling replied:
<img width="410" alt="image" src="https://github.com/TAMULib/scholars-discovery/assets/8352733/8a4dded0-c402-450e-93ea-4138a004427a">

LGTM

tamu-sad-iii committed:
Post demo cleanup

Minor format cleanup and fixed display view yaml

Update README.md

Move redundant code to abstract

Use available static constant from dependency

Update README.md

wwelling commented on README.md:
````diff
@@ -54,24 +54,38 @@ Solr is configured via ```spring.data.solr```.
+
+or
+
+Copy `solr/configsets/scholars-discovery` to local Solr installation and run
````
This alternative development setup provides better performance but requires setup and maintenance locally.

tamu-sad-iii committed:
Further cleanup of docx export and corrected term query for ids

tamu-sad-iii committed:
Zip docx export handlebar helpers for publications categorized by type

tamu-sad-iii committed:
Minor refactor, handlebar helper documentation, updated zip docx export filter

tamu-sad-iii committed:
Fix individual JSON links

tamu-sad-iii committed:
Add patent to zip docx profile

rmathew1011 commented on src/main/resources/defaults/displayViews/defaults/mainContentTemplate.html:
```diff
@@ -9,6 +9,6 @@
-    <span class="ml-1"><a href="{{_links.self.href}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
+    <span class="ml-1"><a href="{{serviceUrl}}/individual/{{id}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
   </span>
 </div>
```
New line could be added.

wwelling replied:
updated

rmathew1011 commented on src/main/resources/defaults/displayViews/processes/mainContentTemplate.html:
```diff
@@ -3,6 +3,6 @@
-    <span class="ml-1"><a href="{{_links.self.href}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
+    <span class="ml-1"><a href="{{serviceUrl}}/individual/{{id}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
   </span>
 </div>
```
Newline could be added.

rmathew1011 commented on src/main/resources/defaults/displayViews/relationships/mainContentTemplate.html:
```diff
@@ -3,6 +3,6 @@
-    <span class="ml-1"><a href="{{_links.self.href}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
+    <span class="ml-1"><a href="{{serviceUrl}}/individual/{{id}}" title="View individual in JSON format" download target="_blank"><i class="fa fa-link"><span class="d-none">individual record</span></i></a></span>
   </span>
 </div>
```
Newline could be added.

tamu-sad-iii committed:
Adjusted zip docx profile publications filter precision and range

tamu-sad-iii committed:
Add missing endlines before eof to template files updated

rmathew1011 commented on src/main/resources/defaults/displayViews/collections/mainContentTemplate.html:
```diff
@@ -22,4 +22,4 @@
     font-weight: 550;
     margin-right: 2px;
   }
-</style>
```
Newline could be added.
